lib/node_modules/@stdlib/_tools/eslint/rules/tsdoc-declarations-doctest/lib/main.js

```diff
@@ -0,0 +1,498 @@
+
+/**
+* Gets or creates a cached regex for finding return annotations.
+*
+* @private
+* @param {string} annotationType - type of annotation (returns, throws, etc.)
+* @param {string} value - expected value
+* @returns {RegExp} compiled regular expression
+*/
+function getAnnotationRegex( annotationType, value ) {
+	var escapedValue;
+	var pattern;
+	var key;
+
+	escapedValue = escapeRegExpString( value );
+	key = annotationType + '::' + escapedValue;
+
+	if ( !regexCache[ key ] ) {
+		pattern = '// ' + annotationType + '\\s+' + escapedValue;
+		regexCache[ key ] = new RegExp( pattern );
+	}
+	return regexCache[ key ];
+}
+
+/**
+* Counts the number of lines in the given string.
+*
+* @private
+* @param {string} str - input string
+* @returns {number} number of lines
+*/
+function countLines( str ) {
+	var lines;
+
+	// Use cached result if available
+	if ( lineCountCache[ str ] ) {
+		return lineCountCache[ str ];
+	}
+	lines = str.split( reEOL() );
+	lineCountCache[ str ] = lines.length - 1;
+	return lineCountCache[ str ];
+}
+
+/**
+* Searches for variable in scope matching the expected value.
+*
+* @private
+* @param {Object} scope - VM scope
+* @param {*} expected - expected value to search for
+* @returns {string} variable name or `?` if no match found
+*/
+function findName( scope, expected ) {
+	var keys = objectKeys( scope );
+	var key;
+	var i;
+	for ( i = 0; i < keys.length; i++ ) {
+		key = keys[ i ];
+		if (
+			!contains( SCOPE_DEFAULTS, key ) &&
+			compareValues( scope[ key ], expected ) === null
+		) {
+			return key;
+		}
+	}
+	return '?';
+}
+
+/**
+* Resolves the implementation path from package.json.
+*
+* @private
+* @param {string} filepath - TypeScript declaration file path
+* @returns {(string|null)} resolved implementation path or null if not found
+*/
+function resolveImplementationPath( filepath ) {
+	var mainPath;
+	var pkgJSON;
+	var pkgPath;
+	var pkgDir;
+	var opts;
+
+	// Find the nearest package.json:
+	opts = {
+		'dir': dirname( resolve( filepath ) )
+	};
+	pkgPath = resolveParentPath( 'package.json', opts );
+	if ( !pkgPath ) {
+		debug( 'Could not find package.json for: ' + filepath );
+		return null;
+	}
+
+	// Read and parse package.json:
+	try {
+		pkgJSON = JSON.parse( readFileSync( pkgPath, 'utf8' ) );
+	} catch ( err ) {
+		debug( 'Could not read/parse package.json: ' + pkgPath + '. Error: ' + err.message );
+		return null;
+	}
+
+	// Get the main entry point:
+	mainPath = pkgJSON.main || './lib';
+	pkgDir = dirname( pkgPath );
+	return resolve( pkgDir, mainPath );
+}
+
+/**
+* Cleans TSDoc comment by removing comment prefixes.
+*
+* @private
+* @param {string} comment - TSDoc comment
+* @returns {string} cleaned comment
+*/
+function cleanTSDocComment( comment ) {
+	// Remove opening /** and closing */
+	var cleaned = replace( replace( comment, /^\/\*\*/, '' ), /\*\/$/, '' );
+
+	// Remove * at the beginning of lines
+	cleaned = replace( cleaned, RE_COMMENT_PREFIX, '' );
+	return cleaned;
```

Comment:
```suggestion
	return replace( cleaned, RE_COMMENT_PREFIX, '' );
```

Intermediate is not necessary.